packages/arcgis-rest-request/src/request.ts

```diff
@@ -295,6 +305,12 @@ export function request(
           // the consumer specified a maximum length for URLs
           // and this would exceed it, so use post instead
           fetchOptions.method = "POST";
+
+          // Add token back to body with other params instead of header
+          if (token.length && options.secureToken) {
+            params.token = token;
+            delete requestHeaders["X-Esri-Authorization"];
+          }
```

Comment:
I think you are right to move the token out of the header and into the body if we have to use POST.

I only wonder if we should check for the existence of the token in the header rather than the re-check the criteria for putting it there.

```suggestion
          const headerToken = requestHeaders["X-Esri-Authorization"];
          if (headerToken) {
            // Add token back to body with other params instead of header
            params.token = headerToken;
            delete requestHeaders["X-Esri-Authorization"];
          }
```

Reply:
@tomwayson We can do that but it will get more complicated. When passed as a header, the value is different. It's actually `Bearer ${token}`. Thought it would be better to use our original which we know is not modified elsewhere.

Reply:
let's leave it as is